### PR TITLE
Adding documentation for filter search in OpenSearch

### PR DESCRIPTION
**Pull request by leanneeliatra**
### Description
Adding a new page to the documentation for **Filtering Search Results** in OpenSearch

### Issues Resolved
Part of https://github.com/opensearch-project/documentation-website/issues/7507

### Version
all

### Frontend features
n/a

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/filter-search.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+---
+layout: default
+title: Filter search results
+nav_order: 36
+---
+
+# Filter search results
+
+In OpenSearch, filtering search results can be achieved through two main approaches: using a [DSL boolean query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html) with a filter clause. The boolean query filtering approach applies filters to both search hits and aggregations.
```
Comment (hdhalter) [inline _search-plugins/filter-search.md:9]:
```suggestion
In OpenSearch, filtering search results can be achieved using a DSL Boolean query with a filter clause. The boolean query filtering approach applies filters to both search hits and aggregations.
```

**Comment by leanneeliatra** [comment]
This is ready for review please cc @hdhalter

**Comment by vagimeli** [comment]
@jainankitk @msfroh @kaushalmahi12 @sgup432 Will one of you provide technical review and approval of this PR? Following your review, we'll move this forward to get it published. Thank you for your support.

**Review by vagimeli — APPROVED** [review]
Doc review complete. Moving to editorial once tech review approves.

**Comment by vagimeli** [comment]
> @jainankitk @msfroh @sgup432 @kaushalmahi12 Will one of you provide technical review and approval of this PR? Following your review, we'll move this forward to get it published. Thank you for your support.

@jainankitk @msfroh @sgup432 @kaushalmahi12 I've completed the doc review and tested the examples using Dev Tools. We need one of you to approve for technical accuracy before we can move this doc forward for publishing. Please review by week's end. Thank you :)

**Comment by msfroh** [comment]
I took a quick look last week while at OpenSearchCon.

My main concern is that the article is covering a variety of topics, between aggregations, filtering, and rescoring. I don't  understand what the goal is -- if it's just filtering search results, one could provide an explanation of the different sub-clauses of a `bool` query. I don't understand how rescoring fits in with the rest of the content.

I think there could be a clearer story around the query level filtering (i.e. a `filter` clause under a `bool` query), filtering aggregations (or not), and applying a `post_filter` on search hits without impacting aggregations. In particular, I believe that `post_filter` makes it easier to implement multi-select for facets in a UI. 

Say, for example, that you show multiple brands (with their hit counts) on your UI and a user clicks on "Brand A". If you were to add that as a `filter` clause to your `bool` query and rerun it, then you would only see results for "Brand A". Assuming you reloaded the page, you would no longer even see "Brand B" as an option (since the results are already filtered). By applying the brand filter as a post filter, you're still able to show all the brands and their counts, allowing users to select multiple brands (and add them all as a disjunction in the `post_filter`).

With the aggregation-level filtering, those can make sense when you do want to filter one set of facets by selections applied to a different set of facets. To continue the example above, if a user has checked brands A and B under the brand facets, I would not want to apply that filter to the top-level query or to the brand facets (since I want to see all possible brands), but I might want to refresh the price facets, since the count of results in different price ranges is affected by the brands I chose. In this case, I would probably add a filter like `{"terms": { "brand": ["Brand A", "Brand B"]}}` to the price range aggregation.

@leanneeliatra -- was that the intention of the page? 

If so, I think it could be conveyed with a story of this hypothetical shopper's journey with three sections:

1. User starts by searching for "smartphone" in category "electronics". These get applied as top-level filters (maybe with `smartphone` as a `must` clause, since you do want it to affect scoring). You show results with brand and price facets.
2. User checks one brand facet. If you added it to the top-level filter, you would filter the results from the "brand" aggregation, losing the rest. Therefore, you add it to `post_filter` instead.
3. When the user selects one or more brand, you want to update the "price" facet based on the brand selection, since it impacts the count of results per price range. This is where you add a `filter` clause to your aggregation. Similarly, if the user selects a price range, you would want to filter the brand aggregation.

It covers three filter types with examples of when and why you would want to use each.

**Comment by leanneeliatra** [comment]
@msfroh Thanks for the feedback! I'm working on it now. I'll update the page to ensure article tells a clearer story. I think the example given flows much better so I'll adjust to the shopper's journey. I’ll keep you posted once I’ve made the updates!

**Comment by leanneeliatra** [comment]
Hi @msfroh, the review comments have been addressed, apologies for the delay in getting them back to you. 

Please re-review and let me know if you have any further suggestions. I appreciate your update. The examples demonstrate filtering in a more focused manner with improved focus for the page, so much appreciated.

**Review by msfroh — APPROVED** [review]
Thanks, @leanneeliatra! This is a great set of examples and tells a really nice story.

**Comment by vagimeli** [comment]
> Thanks, @leanneeliatra! This is a great set of examples and tells a really nice story.

Thanks @leanneeliatra @msfroh. This PR is scheduled for editorial review.

File: _search-plugins/filter-search.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+---
+layout: default
+title: Filter search results
+nav_order: 36
+---
+
+# Filter search results
+
+You can filter search using different methods, each suited to specific scenarios. We can apply filters at the query level, using `boolean` query clauses, as well as using `post_filter` and `aggregation` level filters.
```
Comment (vagimeli) [inline _search-plugins/filter-search.md:9]:
```suggestion
You can filter search using different methods, each suited to specific scenarios. You can apply filters at the query level, using `boolean` query clauses, as well as by using `post_filter` and `aggregation` level filters.
```

File: _search-plugins/filter-search.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+layout: default
+title: Filter search results
+nav_order: 36
+---
+
+# Filter search results
+
+You can filter search using different methods, each suited to specific scenarios. You can apply filters at the query level, using `boolean` query clauses, as well as by using `post_filter` and `aggregation` level filters.
+
+## Query-level filtering with boolean queries
```
Comment (vagimeli) [inline _search-plugins/filter-search.md:11]:
```suggestion
## Query-level filtering with boolean queries

```

File: _search-plugins/filter-search.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
```
Comment (vagimeli) [inline _search-plugins/filter-search.md]:
```suggestion
## Query-level filtering with Boolean queries
```

Comment (vagimeli) [inline _search-plugins/filter-search.md]:
```suggestion
## Query-level filtering with Boolean queries
```

File: _search-plugins/filter-search.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  }
+}
+```
+{% include copy-curl.html %}
+
+Your result should show `BrandA` smartphones in the search hits and all brands in the aggregations.
+
+## Refining aggregrations with aggregation-level filtering
+
+You can use aggregation-level filtering to apply filters to specific aggregations, without affecting the overall aggregation. For example, use aggregation-level filtering to filter the `price_ranges` aggregation based on selected brands, `BrandA` and `BrandB`, without affecting the main `price_ranges` aggregation, as shown in the following example request. This displays price ranges relevant to the selected brands while also displaying overall price ranges for all products.
```
Comment (vagimeli) [inline _search-plugins/filter-search.md:108]:
```suggestion
You can use aggregation-level filtering to apply filters to specific aggregations, without affecting the overall aggregation. 

For example, use aggregation-level filtering to filter the `price_ranges` aggregation based on selected brands, `BrandA` and `BrandB`, without affecting the main `price_ranges` aggregation, as shown in the following example request. This displays price ranges relevant to the selected brands while also displaying overall price ranges for all products.
```

File: _search-plugins/filter-search.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+- **Post-filter filtering:** Use `post_filter` to refine search hits based on user selections while preserving all aggregation options.
+- **Aggregation-level filtering:** Adjust specific aggregations based on selected filters without impacting other aggregations.
+
+## Query-level filtering with Boolean queries
+
+Use a `boolean` query with a filter clause to apply filters to both search hits and aggregations. For example, if a shopper searches for `smartphones` from `BrandA`, a Boolean query can restrict results to only those smartphones from `BrandA`. The following steps guide you through query-level filtering. 
+
+1. Create an index `electronics` and provide the mapping using the following request:
+
+```
```
Comment (natebower) [inline _search-plugins/filter-search.md:21]:
```suggestion
```json
```

File: _search-plugins/filter-search.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      "features": { "type": "keyword" }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+2. Add documents to the `electronics` index using the following request:
+
+```
```
Comment (natebower) [inline _search-plugins/filter-search.md:38]:
```suggestion
```json
```

File: _search-plugins/filter-search.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+  "category": "Smartphone",
+  "price": 799.99,
+  "features": ["5G", "Triple Camera"]
+}
+```
+{% include copy-curl.html %}
+
+3. Apply a `boolean` filter query to display only `smartphones` from `BrandA` using the following request:
+
+```
```
Comment (natebower) [inline _search-plugins/filter-search.md:65]:
```suggestion
```json
```

File: _search-plugins/filter-search.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Narrowing results using `post-filter` while preserving aggregation visibility
+
+Use `post_filter` to limit search hits while preserving all aggregation options. For example, if a shopper selects `BrandA`, results are filtered to show only `BrandA` products while maintaining visibility of all brand options in the aggregations, as shown in the following example request:
```
Comment (natebower) [inline _search-plugins/filter-search.md:82]:
```suggestion
Use `post_filter` to limit search hits while preserving all aggregation options. For example, if a shopper selects `BrandA`, results are filtered to show only `BrandA` products while maintaining the visibility of all brand options in the aggregations, as shown in the following example request:
```

File: _search-plugins/filter-search.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+```
```
Comment (natebower) [inline _search-plugins/filter-search.md:84]:
```suggestion
```json
```

File: _search-plugins/filter-search.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    }
+  },
+  "post_filter": {
+    "term": { "brand": "BrandA" }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Your result should show `BrandA` smartphones in the search hits and all brands in the aggregations.
```
Comment (natebower) [inline _search-plugins/filter-search.md:104]:
```suggestion
The result should show `BrandA` smartphones in the search hits and all brands in the aggregations.
```

File: _search-plugins/filter-search.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+## Refining aggregrations with aggregation-level filtering
```
Comment (natebower) [inline _search-plugins/filter-search.md:106]:
```suggestion
## Refining aggregations with aggregation-level filtering
```

File: _search-plugins/filter-search.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+You can use aggregation-level filtering to apply filters to specific aggregations, without affecting the overall aggregation.
```
Comment (natebower) [inline _search-plugins/filter-search.md:108]:
```suggestion
You can use aggregation-level filtering to apply filters to specific aggregations without affecting the overall aggregation. 
```

File: _search-plugins/filter-search.md [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+}
+```
+{% include copy-curl.html %}
+
+Your result should show `BrandA` smartphones in the search hits and all brands in the aggregations.
+
+## Refining aggregrations with aggregation-level filtering
+
+You can use aggregation-level filtering to apply filters to specific aggregations, without affecting the overall aggregation. 
+
```
Comment (natebower) [inline _search-plugins/filter-search.md:109]: Above:  "without affecting the main aggregation to which they belong"?

File: _search-plugins/filter-search.md [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+For example, use aggregation-level filtering to filter the `price_ranges` aggregation based on selected brands, `BrandA` and `BrandB`, without affecting the main `price_ranges` aggregation, as shown in the following example request. This displays price ranges relevant to the selected brands while also displaying overall price ranges for all products.
```
Comment (natebower) [inline _search-plugins/filter-search.md:110]:
```suggestion
For example, you can use aggregation-level filtering to filter the `price_ranges` aggregation based on selected brands, `BrandA` and `BrandB`, without affecting the main `price_ranges` aggregation, as shown in the following example request. This displays price ranges relevant to the selected brands while also displaying overall price ranges for all products.
```

File: _search-plugins/filter-search.md [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+
+```
```
Comment (natebower) [inline _search-plugins/filter-search.md:112]:
```suggestion
```json
```

**Review by natebower — COMMENTED** [review]
@leanneeliatra @vagimeli Please see my comment and changes and let me know if you have any questions. Thanks!

**Pull request merged**